### PR TITLE
Feature: filter search

### PR DESCRIPTION
**Pull request by gabryelreyes**
Added filtering of search queries by specifiying which fields are to be retrieved and printed.

### CODE REVIEW

File: src/pyJiraCli/cmd_search.py [hunk 1]
```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
...
+
+    if fields == []:
+        fields = DEFAULT_FIELDS
+
+    issues = search_dict['issues']
+    number_of_fields = min(len(fields), MAX_FIELDS_PRINTED)
+
+    print(f"{'Key':<{COLUMN_WIDTH}}", end="")
+
+    for field in range(number_of_fields):
```
Comment (jkerpe) [inline src/pyJiraCli/cmd_search.py:227]: wouldn't it be something like:
`for idx in range(number_of_fields): ...`

File: src/pyJiraCli/cmd_search.py [hunk 2]
```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
...
-        print(f"{issue.fields.summary[:HEADER_COL_WIDTH['Summary'] - 2]:<{HEADER_COL_WIDTH['Summary']}}",
-              end="")
-        print(
-            f"{issue.fields.created[:10]:<{HEADER_COL_WIDTH['Created']}}", end="")
-        print(
-            f"{issue.fields.creator.name:<{HEADER_COL_WIDTH['Creator']}}", end="\n")
+        issue_key = issue['key']
+        print(f"{issue_key:<{COLUMN_WIDTH}}", end="")
+
+        for key in range(number_of_fields):
```
Comment (jkerpe) [inline src/pyJiraCli/cmd_search.py:235]: see above

File: src/pyJiraCli/cmd_search.py [hunk 3]
```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
...
+                try:  # Try to convert the field value to a datetime object
+                    res = datetime.datetime.strptime(
+                        field_value, "%Y-%m-%dT%H:%M:%S.%f%z")
+                    field_value = res.strftime("%Y-%m-%d %H:%M:%S")
+                except ValueError:
+                    # Do nothing. The field value is not a datetime object.
+                    pass
+
+            if len(field_value) > COLUMN_WIDTH-1:
+                field_value = field_value[:COLUMN_WIDTH-4] + "..."
```
Comment (jkerpe) [inline src/pyJiraCli/cmd_search.py:251]: maybe add a comment what is done here. I don't really get the numbers

File: src/pyJiraCli/cmd_search.py [hunk 4]
```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
...
-            f"{issue.fields.creator.name:<{HEADER_COL_WIDTH['Creator']}}", end="\n")
+        issue_key = issue['key']
+        print(f"{issue_key:<{COLUMN_WIDTH}}", end="")
+
+        for key in range(number_of_fields):
+            field = fields[key]
+            field_value = issue["fields"][field]
+
+            if not isinstance(field_value, str):
+                field_value = "Complex Type:" + type(field_value).__name__
```
Comment (jkerpe) [inline src/pyJiraCli/cmd_search.py:240]: is it really desired, that the table now shows `Complex Type:dict` for the fields `creator` and `project` when I use the default fields?

**Review by jkerpe — CHANGES_REQUESTED** [review]

File: src/pyJiraCli/cmd_search.py (hunk 1 above)
Comment (LuDubies) [reply src/pyJiraCli/cmd_search.py:227]: field_idx kombiniert eure Ansätze.

File: src/pyJiraCli/cmd_search.py (hunk 4 above)
Comment (gabryelreyes) [reply src/pyJiraCli/cmd_search.py:240]: If we start handling every complex field, then the tool stops being generic. But I will try to do something for such fields

**Review by jkerpe — APPROVED** [review]

**Pull request merged**